transformations/word2vec_genderify/transformation.py

```diff
@@ -0,0 +1,266 @@
+        noun) else replacement
+    # some words from w2v come out with _, ie. "brain_plasticity", split them apart
+    split_words = replacement.split("_")
+    logging.debug(
+        f"replacing {noun} with {replacement}. and split words are {split_words}"
+    )
+    # if the noun is a plural one, convert the words to the plural form
+    if not is_singular_noun:
+        # in english typically only the last word in a compound noun is pluralized
+        split_words[-1] = inflect_engine.plural(split_words[-1])
+    replacement = " ".join(split_words)
+    return replace_word_in_sentence(sentence, noun, replacement)
+
+
+def generate_sentences(sentence: str, max_outputs: int, wv, nlp,
+                       inflect_engine) -> List[str]:
+    """take a sentence and return a list of sentences with each proper noun replaced
+
+    Args:
+        sentence ([str]): sentence
+        max_outputs (int) : max sentences to return
+        wv ([gensim.models.keyedvectors.KeyedVectors]): word2vec gensim model
+        nlp ([spacy.lang]): spacy nlp object
+        inflect_engine : engine from inflect module that allows easy pluralization and singularization
+    
+    Returns:
+        [list]: list of sentences with each noun replaced with a male-ify + a woman-ified version
+    """
+    nouns = get_nouns(sentence, nlp)
+
+    banned_words = ["man", "woman", "men", "women", "boy", "girl"]
+    manify_function = lambda word: x_ify(word=word,
+                                         wv=wv,
+                                         x=('man', wv['man']),
+                                         inflect_engine=inflect_engine,
+                                         banned_words=banned_words)
+    womanify_function = lambda word: x_ify(word=word,
+                                           wv=wv,
+                                           x=('woman', wv['woman']),
+                                           inflect_engine=inflect_engine,
+                                           banned_words=banned_words)
+    swapped_sentences = []
+    count = 0
+    for noun in nouns:
+        if noun.lower(
+        ) in banned_words:  # ignore replacing man/woman with itself
+            continue
+        swapped_sentence = swap_out_words(sentence, str(noun), manify_function,
+                                          inflect_engine)
+        if sentence != swapped_sentence:  # only add sentence if its a new sentence
+            swapped_sentences.append(swapped_sentence)
+            count += 1
+        swapped_sentence = swap_out_words(sentence, str(noun),
+                                          womanify_function, inflect_engine)
+        if sentence != swapped_sentence:  # only add if there are any changes
+            swapped_sentences.append(swapped_sentence)
+            count += 1
+        if count > max_outputs:
+            break
+    if count == 0:
+        return [sentence]
+    return swapped_sentences[:max_outputs]
```

Comment:
Here `manify_function` is prioritized to `womanify_function`.
For example, if `max_outputs ` = 1, and `swapped_sentence ` from `manify_function` is different from the original sentence, the code always outputs that, without chance of selecting a possible  `swapped_sentence ` from `womanify_function `. I suggest to randomize this.